lib/logstash/inputs/http.rb

```diff
@@ -236,19 +241,37 @@ def validate_ssl_settings!
     end
 
     if @ssl && (original_params.key?("verify_mode") && original_params.key?("ssl_verify_mode"))
-        raise LogStash::ConfigurationError, "Both 'ssl_verify_mode' and 'verify_mode' were set. Use only 'ssl_verify_mode'."
+      raise LogStash::ConfigurationError, "Both `ssl_verify_mode` and (deprecated) `verify_mode` were set. Use only `ssl_verify_mode`."
     elsif original_params.key?("verify_mode")
       @ssl_verify_mode_final = @verify_mode
-    elsif original_params.key?("ssl_verify_mode")
-      @ssl_verify_mode_final = @ssl_verify_mode
     else
       @ssl_verify_mode_final = @ssl_verify_mode
     end
 
+    if @ssl && (original_params.key?('cipher_suites') && original_params.key?('ssl_cipher_suites'))
+      raise LogStash::ConfigurationError, "Both `ssl_cipher_suites` and (deprecated) `cipher_suites` were set. Use only `ssl_cipher_suites`."
+    elsif original_params.key?('cipher_suites')
+      @ssl_cipher_suites_final = @cipher_suites
+    else
+      @ssl_cipher_suites_final = @ssl_cipher_suites
+    end
+
+    if @ssl && (original_params.key?('tls_min_version') && original_params.key?('ssl_supported_protocols'))
+      raise LogStash::ConfigurationError, "Both `ssl_supported_protocols` and (deprecated) `tls_min_ciphers` were set. Use only `ssl_supported_protocols`."
+    elsif @ssl && (original_params.key?('tls_max_version') && original_params.key?('ssl_supported_protocols'))
+      raise LogStash::ConfigurationError, "Both `ssl_supported_protocols` and (deprecated) `tls_max_ciphers` were set. Use only `ssl_supported_protocols`."
+    else
+      if @ssl && (original_params.key?('tls_min_version') || original_params.key?('tls_max_version'))
+        @ssl_supported_protocols_final = TLS.get_supported(tls_min_version..tls_max_version).map(&:name)
+      else
+        @ssl_supported_protocols_final = @ssl_supported_protocols
+      end
+    end
```

Comment:
Same observation as the one above, protect with an `if @ssl` could make it more readable.

Reply:
if the attr is not set `@ssl_supported_protocols_final = @ssl_supported_protocols` is just `nil`

Reply:
here's a compromise refactoring: https://github.com/logstash-plugins/logstash-input-http/pull/151/commits/db159bbfe07de78546bb3f2430fd0e616a181249 ... wdyt?